src/compiler/es-next.js

```diff
@@ -0,0 +1,145 @@
+import { dirname, join, relative, sep as pathSep } from 'path';
+import { readFileSync } from 'fs';
+import stripBom from 'strip-bom';
+import sourceMapSupport from 'source-map-support';
+import nodeVer from 'node-version';
+import Globals from '../api/globals';
+
+const COMMON_API_PATH   = join(__dirname, '../api/common');
+const NODE_MODULES_PATH = join(__dirname, '../../node_modules');
+const NODE_VER          = parseInt(nodeVer.major, 10);
+const CWD               = process.cwd();
+
+const FIXTURE_RE = /(^|;|\s+)fixture\s*(\(\s*('|").+?\3\s*\)|`.+?`)/;
+const TEST_RE    = /(^|;|\s+)test\s*(\(\s*('|").+?\3\s*,)/;
+
+var Module = module.constructor;
+
+export default class ESNextCompiler {
+    constructor () {
+        this.sourceMaps = {};
+        this.cache      = {};
+
+        this._setupSourceMapsSupport();
+    }
+
+    static _getNodeModulesLookupPath (filename) {
+        var dir   = dirname(filename);
+        var paths = Module._nodeModulePaths(dir);
+
+        // HACK: for npm v2 installations we need to expose our node_modules,
+        // so module will be able to require babel-runtime.
```

Comment:
```
        // HACK: for npm v2 installations, we need to expose our node_modules,
        // so that Module will be able to require babel-runtime.
```